react-web/src/hooks/useLogs.ts

```diff
@@ -13,41 +14,119 @@ export const useLogs = (
     testEnded: boolean,
     handleErrorScenario: () => void
 ) => {
+    const dispatch = useDispatch()
     const [logInfo, setLogInfo] = useState<Log[]>([])
     const [fetchingLogs, setFetchingLogs] = useState(false);
     const [refetchLogsOffset] = useState(1);
+    const [enabled, setEnabled] = useState(false)
+
+    const { startTime, endTime, runState, ended } = useAppSelector((state) => state.runTime);
+
+    useEffect(() => {
+        if (testEnded) { 
+            dispatch(setEnded(ended+1)); 
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [testEnded])
+
+
+    useEffect(() => {
+        setEnabled(!fetchingLogs && !(ended > 1) && !!uuid)
+    }, [fetchingLogs, ended, uuid])
+
     //reset log if uuid changed
     useEffect(()=>{
         setLogInfo([])
+        dispatch(setEnded(0))
     },[uuid])
 
-    const fetchLog = React.useCallback(async ()=>{
+    const captureRunTime = (sTime: string, eTime: string, state: string) => {
+        if (sTime && eTime && state) { 
+            dispatch(setStates({startTime: sTime, endTime: eTime, runState: state}))
+            dispatch(setBuildInfo())
+        }
+    }
+
+    const computeRunState = (state: string) => {
+        if (state.indexOf('build-flake') !== -1) { 
+            return 'building'
+        } else if (state.indexOf('generate-flake') !== -1) {
+            return 'preparing'
+        } else if (state.indexOf('run-certify') !== -1) {
+            return 'certifying'
+        } else {
+            return '' ;
```

Comment:
mixing rule of using `;` with other places of not using; I strongly recommend starting the use of `eslint` and enforcing some consistent set of rules

Reply:
Corrected this instance. Will ensure in future, as well as correct the other places with time.